src/components/WalletConnect/NanoContract/NanoContractActions.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+} from 'react-native';
+import { t } from 'ttag';
+import { HathorFlatList } from '../../HathorFlatList';
+import { commonStyles } from '../theme';
+import { getShortHash, renderValue } from '../../../utils';
+import { ReceivedIcon } from '../../Icons/Received.icon';
+import { SentIcon } from '../../Icons/Sent.icon';
+import { COLORS } from '../../../styles/themes';
+import { DEFAULT_TOKEN } from '../../../constants';
+
+/**
+ * It returns the title template for each action type,
+ * which is either 'deposit' or 'withdrawal'.
+ *
+ * @param {string} tokenSymbol The token symbol fetched from metadata,
+ * or a shortened token hash.
+ *
+ * @returns {string} A title template by action type.
+ */
+const actionTitleMap = (tokenSymbol) => ({
+  deposit: t`${tokenSymbol} Deposit`,
+  withdrawal: t`${tokenSymbol} Withdrawal`,
+});
+
+/**
+ * Get action title depending on the action type.
+ * @param {Object} tokens A map of token metadata by token uid
+ * @param {Object} action An action object
+ *
+ * @example
+ * getActionTitle({ '123': { ..., symbol: 'STR' }}, { ..., token: '123', type: 'deposit' })
+ * >>> 'STR Deposit'
+ *
+ * @example
+ * getActionTitle({}, { ..., token: '1234...5678', type: 'deposit' })
+ * >>> '1234...5678 Deposit'
+ */
+const getActionTitle = (tokens, action) => {
+  const tokenMetadata = tokens[action.token];
+  if (tokenMetadata) {
+    return actionTitleMap(tokenMetadata.symbol)[action.type];
+  }
+
+  if (action.token === DEFAULT_TOKEN.uid) {
+    return actionTitleMap(DEFAULT_TOKEN.symbol)[action.type]
+  }
+
+  return actionTitleMap(getShortHash(action.token))[action.type];
+};
+
+/**
+ * It renders a list of actions with a proper title for each one.
+ * It renders nothing if there aren't actions to render.
+ *
+ * @param {Object} props
+ * @param {Object[]} props.ncActions A list of Nano Contract actions.
+ * @param {Object} props.tokens A map of token metadata by token uid.
+ */
+export const NanoContractActions = ({ ncActions, tokens }) => {
+  if (!ncActions || ncActions.length < 1) {
+    return null;
+  }
+
+  const styles = StyleSheet.create({
+    wrapper: { marginTop: 0, marginBottom: 0, marginHorizontal: 0 },
+    action: [commonStyles.text, commonStyles.bold],
+    valueLabel: [commonStyles.text, commonStyles.value, commonStyles.bold, commonStyles.mb4],
+    value: [commonStyles.text, commonStyles.value],
+  });
+
+  return (
+    <View>
+      <View>
+        <Text style={commonStyles.sectionTitle}>{t`Action List`}</Text>
+      </View>
+      <HathorFlatList
+        scrollEnabled={false} // it avoids nearest scrolls
```

Comment:
I've used `HathorFlatList` just for visual consistency, otherwise I would have to build another component only for this list. I don't know for sure what happens if we have 30 actions, I will try here and check. But `scrollEnabled=false` avoids pass the parent scroll control to the this component. It means if the user tap over an action and swipe up or down, the control will still be kept in the parent component, thus moving the parent pane.

Reply:
A large list of actions would just expand the component without create a scroll context in the child component, which is what I was expecting, and that's good. See:

<img src="https://github.com/user-attachments/assets/e35abc8b-c11e-405a-bc27-5034663f9fd8" height="600" />